src/comments.js

```diff
@@ -834,7 +834,33 @@ function printComment(commentPath, options) {
       return "#" + comment.value;
     case "CommentBlock":
     case "Block":
-      return "/*" + comment.value + "*/";
+      const lines = comment.value.split("\n");
+      const isJsDocComment =
+        lines.length > 1 &&
+        lines.slice(0, lines.length - 1).every(line => line.trim()[0] === "*");
+      if (!isJsDocComment) {
+        return "/*" + comment.value + "*/";
+      }
+
+      // Determine the indentation level of the parent node.
+      const parentNode = commentPath.getParentNode();
+      const indentLevel = parentNode.loc.start.column;
+
+      let spaces = "";
+      for (let i = 0; i <= indentLevel; i++) {
+        spaces += " ";
+      }
```

Comment:
This won't work as intended when using tabs. We already have logic to figure out indentation and replay it for the template literal use case: https://github.com/prettier/prettier/blob/e378f5e9baf2865b0ed6fef445f29bb3c47c2ccd/src/printer.js#L1855-L1867

I would extract it out into a reusable function and use it here.